### ISSUE TITLE
Segfault in ws-daemon

### ISSUE THREAD

**Issue by meysholdt** [gitpod-io/gitpod#5244]
Spottet twice in prod ([here](https://console.cloud.google.com/logs/query;cursorTimestamp=2021-08-17T18:21:45.089600007Z;query=%22invalid%20memory%20address%22%0Aresource.labels.cluster_name%3D%2528%22gp-prod-ws-eu14-eu-west1%22%20OR%20%22gp-prod-ws-us14-us-west1%22%2529%0Atimestamp%3D%222021-08-17T16:21:40.469677134Z%22%0AinsertId%3D%22gqvs2k8h3qaw0eub%22;summaryFields=resource%252Flabels%252Fcluster_name:false:32:beginning;timeRange=PT6H?project=gitpod-191109) and [here](https://console.cloud.google.com/logs/query;cursorTimestamp=2021-08-17T18:21:45.089600007Z;query=%22invalid%20memory%20address%22%0Aresource.labels.cluster_name%3D%2528%22gp-prod-ws-eu14-eu-west1%22%20OR%20%22gp-prod-ws-us14-us-west1%22%2529%0Atimestamp%3D%222021-08-17T16:21:40.469677134Z%22%0AinsertId%3D%22gqvs2k8h3qaw0eub%22;summaryFields=resource%252Flabels%252Fcluster_name:false:32:beginning;timeRange=PT6H?project=gitpod-191109)) after today's deployment:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1c21260]

goroutine 1 [running]:
github.com/gitpod-io/gitpod/ws-daemon/pkg/dispatch.(*Dispatch).Close(0xc0002d8310, 0x0, 0x0)
	github.com/gitpod-io/gitpod/ws-daemon/pkg/dispatch/dispatch.go:152 +0x100
github.com/gitpod-io/gitpod/ws-daemon/pkg/daemon.(*Daemon).Stop(0xc000532840, 0x0, 0xc0008f8eb8)
	github.com/gitpod-io/gitpod/ws-daemon/pkg/daemon/daemon.go:178 +0x39
```

The call stack looks like the one from https://github.com/gitpod-io/gitpod/pull/5229
Container-version of `ws-daemon`: commit-[2e90d32ba9ece42b8f56d1bb298c3ba53b4d2f99](https://github.com/gitpod-io/gitpod/commit/2e90d32ba9ece42b8f56d1bb298c3ba53b4d2f99).

**Comment by JanKoehnlein** [comment]
Line numbers are not matching anything sensible in this commit. To me it looks like the log is from a state before that the commit fixed.

**Comment by aledbf** [comment]
Closing. Fixed in #5229
